Improbable/Stdlib/Improbable.Stdlib/Platform/Platform.cs

```diff
@@ -0,0 +1,236 @@
+                : Path.Combine(Environment.GetEnvironmentVariable("HOME"), ".improbable"));
+
+        private static PlatformCredential GetCredentials()
+        {
+            var refreshToken = File.ReadAllText(Path.Combine(ToolbeltConfigDir, "oauth2/oauth2_refresh_token"));
+            return new PlatformRefreshTokenCredential(refreshToken);
+        }
+
+        private static PlatformApiChannel GetLocalApiChannel()
+        {
+            return new PlatformApiChannel(GetCredentials(), new PlatformApiEndpoint("localhost", 9876, true));
+        }
+
+        public static Deployment StartLocal(in StartDeploymentOptions startDeployment)
+        {
+            var client = DeploymentServiceClient.Create(GetLocalApiChannel());
+
+            Shell.Run("spatial", "auth", "login", "--json_output");
+
+            // The runtime creates a '.improbable' folder next to the spatialos.json file.
+            var rootDir = Path.GetDirectoryName(startDeployment.ProjectConfigPath);
+            var tempDir = Path.Combine(rootDir, ".improbable");
+
+            Directory.CreateDirectory(tempDir);
+
+            var splConfigPath = Path.Combine(tempDir, "spl_config.json");
+            Shell.Run(
+                "spatial", "alpha", "config", "convert", "--json_output", "--launch_config", Shell.Escape(startDeployment.LaunchConfigPath),
+                "--main_config", Shell.Escape(startDeployment.ProjectConfigPath), "--output_path", splConfigPath);
+
+            var project = JsonConvert.DeserializeObject<ProjectConfig>(File.ReadAllText(startDeployment.ProjectConfigPath, Encoding.UTF8));
+
+            var needsStart = true;
+            var pidFile = Path.Combine(ToolbeltConfigDir, "spatiald.pid");
+            var lockFile = Path.Combine(ToolbeltConfigDir, "spatiald.lock");
+            if (File.Exists(pidFile) && int.TryParse(File.ReadAllText(pidFile).Trim(), out var pid))
+            {
+                try
+                {
+                    Process.GetProcessById(pid);
+                    needsStart = false;
+                }
+                catch
+                {
+                    File.Delete(pidFile);
+
+                    if (File.Exists(lockFile))
+                    {
+                        File.Delete(lockFile);
+                    }
+                }
+            }
+
+            if (needsStart)
+            {
+                Shell.Run("spatial", "service", "start", "--json_output", "--main_config", Shell.Escape(startDeployment.ProjectConfigPath));
+            }
+
+            using (var tcs = new CancellationTokenSource(5000))
+            {
+                var timeout = CallSettings.FromCancellationToken(tcs.Token);
+                var deployments = client.ListDeployments(new ListDeploymentsRequest
+                    { DeploymentName = "local", ProjectName = project.ProjectName }, timeout);
+
+                foreach (var dpl in deployments.Where(dpl => dpl.Status == Deployment.Types.Status.Running))
```

Comment:
You should be able to add a filter onto the `ListDeploymentsRequest`? Or at least you can when using `spot` (which leads to me believe you can with the platform SDK